lib/appsignal/probes/sidekiq.rb

```diff
@@ -31,8 +38,15 @@ def call
       attr_reader :cache
 
       def track_redis_info
-        return unless ::Sidekiq.respond_to?(:redis_info)
-        redis_info = ::Sidekiq.redis_info
+        redis_info = nil
+        if self.class.sidekiq7_and_greater?
+          # Sidekiq >= 7 using the redis-client gem
+          ::Sidekiq.redis { |c| redis_info = c.info }
+        elsif ::Sidekiq.respond_to?(:redis_info)
+          # Sidekiq <= 6 using the redis gem
+          redis_info = ::Sidekiq.redis_info
+        end
+        return unless redis_info
```

Comment:
I may be having a bad case of the architecture astronaut here, so definitely absolutely feel free to ignore this. 🧑‍🚀

Would it make sense to move these differences in how to interact with the Sidekiq Redis to separate classes, and pick one or the other on initialization?

It would avoid having `sidekiq7_and_greater?` checked every time, move these details of how to interact with Redis in different versions to separate functions, and I think it would require less comments explaining what's going on, as the structure of the code would communicate it for you.

```ruby
class Sidekiq7Redis
  def self.info
    ::Sidekiq.redis { |c| redis_info = c.info }
  end
  # And `self.host` for the connection host
end
# And a `Sidekiq6Redis` class implementing the same methods

class SidekiqProbe
  def initialize
    @redis = self.sidekiq7_and_greater? ? Sidekiq7Redis : Sidekiq6Redis
  end
  
  def track_redis_info
    redis_info = @redis.info
    return unless redis_info
  end
end

Reply:
I've applied your suggestion.